controllers/extension-certificate-service/pkg/controller/lifecycle/actuator.go

```diff
@@ -0,0 +1,126 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lifecycle
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/gardener/gardener-extensions/controllers/extension-certificate-service/pkg/controller/lifecycle/internal"
+
+	"github.com/gardener/gardener-extensions/controllers/extension-certificate-service/pkg/imagevector"
+
+	configv1alpha1 "github.com/gardener/gardener-extensions/controllers/extension-certificate-service/pkg/apis/config/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/gardener/gardener-extensions/controllers/extension-certificate-service/pkg/utils"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/rest"
+)
+
+// Actuator acts upon Configuration.
+type Actuator interface {
+	// Reconcile the Configuration.
+	Reconcile(ctx context.Context, config configv1alpha1.Configuration) error
+}
+
+// ActuatorName is the name of the Certificate Service Lifecycle actuator.
+const ActuatorName = "certificate-service-lifecycle-actuator"
+
+type actuator struct {
+	client client.Client
+	config *rest.Config
+
+	logger logr.Logger
+}
+
+// NewActuator returns an actuator responsible for Configuration.
+func NewActuator() Actuator {
+	return &actuator{
+		logger: log.Log.WithName(ActuatorName),
+	}
+}
+
+// Create the Configuration.
+func (a *actuator) Reconcile(ctx context.Context, config configv1alpha1.Configuration) error {
+	return a.DeployCertManager(ctx, config)
+}
+
+// InjectConfig injects the rest config to this actuator.
+func (a *actuator) InjectConfig(config *rest.Config) error {
+	a.config = config
+	return nil
+}
+
+// InjectClient injects the client to this actuator.
+func (a *actuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+const certManagerName = "cert-manager"
+
+var certManagerChartPath = filepath.Join(utils.ChartsPath, certManagerName)
+
+func (a *actuator) DeployCertManager(ctx context.Context, config configv1alpha1.Configuration) error {
+	applier, err := kubernetes.NewChartApplierForConfig(a.config)
```

Comment:
You can instantiate the chart applier only once during `InjectConfig` - saves you error handling + instantiation time.